ext/opentelemetry-exporter-cloud-monitoring/src/opentelemetry/exporter/cloud_monitoring/__init__.py

```diff
@@ -218,12 +226,39 @@ def export(
             # Cloud Monitoring API allows, for any combination of labels and
             # metric name, one update per WRITE_INTERVAL seconds
             updated_key = (metric_descriptor.type, record.labels)
-            last_updated_seconds = self._last_updated.get(updated_key, 0)
-            if seconds <= last_updated_seconds + WRITE_INTERVAL:
+            last_updated_time, _ = self._last_updated.get(updated_key, (0, 0))
+            if seconds <= last_updated_time + WRITE_INTERVAL:
                 continue
-            self._last_updated[updated_key] = seconds
+
+            if (
+                metric_descriptor.metric_kind
+                == MetricDescriptor.MetricKind.CUMULATIVE
+            ):
+                if (
+                    instrument.meter.stateful
+                    or updated_key not in self._last_updated
+                ):
+                    # The aggregation has not reset since the exporter has started up, so that is the start time
+                    point.interval.start_time.seconds = int(
+                        self._exporter_start_time_seconds
+                    )
+                    point.interval.start_time.nanos = int(
+                        self._exporter_start_time_nanos
+                    )
+                else:
+                    # The aggregation reset the last time it was exported
+                    point.interval.start_time.seconds = int(
+                        self._last_updated[updated_key][0]
+                    )
+                    point.interval.start_time.nanos = int(
```

Comment:
Should make a note here about how +1 nanos wasn't enough

Reply:
done

Reply:
Nice, actually saw this in the docs https://cloud.google.com/monitoring/api/ref_v3/rpc/google.monitoring.v3#timeinterval